packages/stacking/src/utils.ts

```diff
@@ -349,3 +359,135 @@ export function ensureLegacyBtcAddressForPox1({
     throw new Error('PoX-1 requires P2PKH/P2SH/P2SH-P2WPKH/P2SH-P2WSH bitcoin addresses');
   }
 }
+
+/**
+ * @internal
+ * Throws if signer args are given for <= PoX-3 or the signer args are missing otherwise.
+ */
+export function ensureSignerArgsReadiness({
+  contract,
+  signerKey,
+  signerSignature,
+  maxAmount,
+  authId,
+}: {
+  contract: string;
+  signerKey?: string;
+  signerSignature?: string;
+  maxAmount?: IntegerType;
+  authId?: IntegerType;
+}) {
+  const hasMaxAmount = typeof maxAmount !== 'undefined';
+  const hasAuthId = typeof authId !== 'undefined';
+  if (/\.pox(-[2-3])?$/.test(contract)) {
+    // .pox, .pox-2 or .pox-3
+    if (signerKey || signerSignature || hasMaxAmount || hasAuthId) {
+      throw new Error(
+        'PoX-1, PoX-2 and PoX-3 do not accept a `signerKey`, `signerSignature`, `maxAmount` or `authId`'
+      );
+    }
+  } else {
+    // .pox-4 or later
+    if (!signerKey || !hasMaxAmount || typeof authId === 'undefined') {
```

Comment:
```suggestion
    if (!signerKey || !hasMaxAmount || !hasAuthId) {
```
Is this logic right?

Reply:
That for `authId` being a number and `0` being falsy in javascript